k8s-charm/lib/charms/juju_dashboard/v0/juju_dashboard.py

```diff
@@ -77,16 +77,27 @@ def __init__(self, data):
                 juju-controller application.
 
         """
+        # Juju controller provides is-juju, while JAAS provides is_juju.
+        is_juju = (
+            data.get("is_juju", True)
+            if data.get("is-juju") is None
+            else data.get("is-juju")
+        )
```

Comment:
Thanks for the review. I've fixed all the issue. Here I think we would still need to leave the default value to be `True` as it was previously assumed to be, so something like this should work:
```
is_juju = data.get("is_juju") or data.get("is-juju", True)
```

Reply:
I think with your suggestion it will fail because when relating to JAAS, `is-juju` will never exist and then the default value of `True` will be used which means the final value can never be `false`.
I think the easiest fix in fact is if we change the JAAS charm to use the same value as the Juju charm. That is easily done. Then you can just check is-juju.
For now though you should be fine setting the default values as `False` for both is_juju and is-juju because the Juju charm already sets their side as `True` [here](https://github.com/juju/juju-controller/blob/3.1/src/charm.py#L43).